code/modules/modular_computers/computers/item/computer_ui.dm

```diff
@@ -112,7 +115,9 @@
 				update_icon()
 				return
 
-			if(idle_threads.len >= processor_unit.max_idle_programs+1)
+			var/obj/item/weapon/computer_hardware/processor_unit/PU = all_components[MC_CPU]
+
+			if(idle_threads.len >= PU.max_idle_programs+1)
```

Comment:
Yup. I Had 0 clue how most of it worked, so I mainly just focused on the interface and nanoui -> tgui stuff.